securedrop/source_app/session_manager.py

```diff
@@ -27,80 +28,76 @@ class UserHasBeenDeleted(_InvalidUserSession):
 
 
 class SessionManager:
-    _CACHE = SimpleCache()  # A mapping of passphrase -> SourceUser
-    _KEY_IN_SESSION_COOKIE = "codename"  # The key in flask.session for the user's passphrase
+    """Helper to manage the user's session cookie accessible via flask.session."""
+
+    # The keys in flask.session for the user's passphrase and expiration date
+    _SESSION_COOKIE_KEY_FOR_CODENAME = "codename"
+    _SESSION_COOKIE_KEY_FOR_EXPIRATION_DATE = "expires"
 
     @classmethod
-    def log_user_in(cls, user: SourceUser, user_passphrase: "DicewarePassphrase") -> None:
+    def log_user_in(
+        cls,
+        db_session: sqlalchemy.orm.Session,
+        supplied_passphrase: "DicewarePassphrase"
+    ) -> SourceUser:
+        # Validate the passphrase; will raise an exception if it is not valid
+        source_user = authenticate_source_user(
+            db_session=db_session, supplied_passphrase=supplied_passphrase
+        )
+
         # Save the passphrase in the user's session cookie
-        session[cls._KEY_IN_SESSION_COOKIE] = user_passphrase
+        session[cls._SESSION_COOKIE_KEY_FOR_CODENAME] = supplied_passphrase
 
-        # Store the user's information
-        cls._CACHE.set(
-            key=user_passphrase, value=user, timeout=config.SESSION_EXPIRATION_MINUTES * 60
-        )
+        # Save the session expiration date in the user's session cookie
+        session_duration = timedelta(minutes=config.SESSION_EXPIRATION_MINUTES)
+        session[cls._SESSION_COOKIE_KEY_FOR_EXPIRATION_DATE] = datetime.utcnow() + session_duration
```

Comment:
I've put back the session expiration date in the session cookie; in my previous PR it was in the local cache.